### PR TITLE
 Dialogue requires Java 11

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
 Dialogue requires Java 11
==COMMIT_MSG==

### CODE REVIEW

File: dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java [hunk 1]
```diff
@@ -235,21 +199,132 @@ static CloseableClient wrap(
                     .clientType(CLIENT_TYPE)
                     .build();
             createMeter.mark();
-            return newInstance;
+            CloseableClient wrapper = new CloseableClientWrapper(newInstance);
+            CleanerSupport.register(wrapper, newInstance::closeApacheClient);
+            return wrapper;
```
Comment (carterkozak) [inline dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java:204]: refactor exists for these three lines -- we must create a delegate which can be strongly referenced by the cleaner without blocking the sentinel reference (the wrapper) from being garbage collected. When the outer class is garbage collected, our cleaner is able to execute.

File: build.gradle [hunk 2]
```diff
@@ -60,7 +60,7 @@ subprojects {
     apply plugin: 'java-library'
     apply plugin: 'org.inferred.processors'
 
-    sourceCompatibility = 1.8
+    sourceCompatibility = 11
```
Comment (jkozlowski) [inline build.gradle:63]: Can you remove overrides in dialogue-annotations-example and dialogue-annotations-processor?

File: dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ResponseLeakDetector.java [hunk 3]
```diff
@@ -71,9 +72,6 @@ Response wrap(Response input, Endpoint endpoint) {
     }
 
     private boolean shouldApplyLeakDetection() {
```
Comment (jkozlowski) [inline dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ResponseLeakDetector.java]: While you're here, can you remove the whole shouldApplyLeakDetection? probability is 1 now, so all this code can probably go away?

File: dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java (hunk 1 above)
Comment (jkozlowski) [reply dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java:204]: Discussed: I found this kinda hard to reason about, the cleaner stuff always is fun. We discussed instead having private static final InternalCloseableClient and public static final class CloseableClient that takes InternalCloseableClient and delegates. I find that easier to understand, and Carter doesn't so let's just ship this.

**Review by jkozlowski — COMMENTED** [review]
I think there is 1 action and maybe another action (but can be in a FLUP PR)

File: dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ResponseLeakDetector.java (hunk 3 above)
Comment (carterkozak) [reply dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ResponseLeakDetector.java]: good catch!

**Review by jkozlowski — APPROVED** [review]
approving to unblock, since I'm going offline shortly

File: build.gradle (hunk 2 above)
Comment (carterkozak) [reply build.gradle:63]: Also removed the `-source 11` lines from the test javac runner since it's the new default.

**Pull request merged**